File: August/Week2/workArea/pascalTriangleII.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
    
    /* naive approach
    let res = Array(rowIndex+1).fill(0);
    // res.fill(0, 0, rowIndex+1);
    res[0]=1;
    // console.log(res);
    
    // res[0]=1;
    for(let i = 1; i<=rowIndex; i++) {
        for(let j = i; j > 0; j--) {
            res[j] = res[j]+res[j-1];
        }
    }
    
    // console.log(res);
    return res;*/
    
    // combinatorics approach
    
    let res = Array(rowIndex+1).fill(0);
    
    res[0]=1;
    
    for(let i=1; i<=rowIndex; i++) {
        res[i] = res[i-1]*parseInt(rowIndex - (i-1))/(i);
        // console.log(parseInt(rowIndex-(i-1)/(i)))
    }
    // console.log(res);
    return res;
    
};

File: August/Week2/Day5/workArea/pascalTriangleII.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        # res = [0]*(rowIndex+1)
        # print(res)
        # res[0]=1
        res = [1]
        for i in range(1, rowIndex+1):
            # print(i)
            # int(triangle[i - 1] * (rowIndex - (i - 1)) / i)
            res.append(int(res[i-1]*(rowIndex - (i-1))/i))
            # res[i] = int(res[i-1] * (rowIndex - (i-1)/i));
            # res[i] = int(res[i])
            # print(int((rowIndex - (i-1))/i))
        # print(res)
        return res
        # return res